clmm/dataops/__init__.py

```diff
@@ -105,30 +96,10 @@ def compute_tangential_and_cross_components(
     geometry: str, optional
         Sky geometry to compute angular separation.
         Options are curve (uses astropy) or flat.
-    is_deltasigma: bool
-        If `True`, the tangential and cross components returned are multiplied by Sigma_crit.
-        Results in units of :math:`M_\odot\ Mpc^{-2}`
-    cosmo: clmm.Cosmology, optional
-        Required if `is_deltasigma` is True and `sigma_c` not provided.
-        Not used if `sigma_c` is provided.
-    z_lens: float, optional
-        Redshift of the lens, required if `is_deltasigma` is True and `sigma_c` not provided.
-        Not used if `sigma_c` is provided.
-    z_source: array, optional
-        Redshift of the source, required if `is_deltasigma` is True and `sigma_c` not provided.
-        Not used if `sigma_c` is provided or `use_pdz=True`.
-    sigma_c : float, optional
-        Critical surface density in units of :math:`M_\odot\ Mpc^{-2}`,
-        if provided, `cosmo`, `z_lens` and `z_source` are not used.
-    use_pdz: bool
-        Flag to use or not the source redshift p(z), required if `is_deltasigma` is True
-        and `sigma_c` not provided. If `False`, the point estimate provided by `z_source` is used.
-    pzpdf : array, optional
-        Photometric probablility density functions of the source galaxies, required if
-        `is_deltasigma=True` and `use_pdz=True` and `sigma_c` not provided.
-    pzbins : array, optional
-        Redshift axis on which the individual photoz pdf is tabulated, required if
-        `is_deltasigma=True` and `use_pdz=True` and `sigma_c` not provided.
+    sigma_c : None, array_like
+        Critical (effective) surface density in units of :math:`M_\odot\ Mpc^{-2}`.
+        If equals to None, shear components are returned, else :math:`\Delta \Sigma` components
+        are returned.
```

Comment:
I like the simplicity brought by having the user previously compute sigma_c, then to pass this to all the functions. But I'm wondering if 'hiding' the decision of computing DeltaSigma instead of shear in the sigma_c keyword may yield some confusion. An option would be to keep a `is_deltasigma` flag for the user to specify, and if `True` check that the user has indeed provided sigma_c. Just a suggestion though.